python/pyarrow/types.pxi

```diff
@@ -579,6 +620,23 @@ cdef class UnionType(DataType):
         for i in range(len(self)):
             yield self[i]
 
+    def field(self, i):
```

Comment:
Is there a test case covering UnionType as well?

Reply:
:+1: